addons/stock/tests/test_packing.py

```diff
@@ -1739,6 +1739,43 @@ def test_compute_hide_picking_type_multiple_records(self):
         self.assertTrue(picking1.hide_picking_type)
         self.assertFalse(picking2.hide_picking_type)
 
+    def test_put_in_pack_partial_different_destinations(self):
+        """ Test putting some of the move lines of a pikcing with different destinations in a package """
+        self.productA.tracking = 'serial'
+
+        picking = self.env['stock.picking'].create({
+            'picking_type_id': self.warehouse.in_type_id.id,
+        })
+        picking_form = Form(picking)
+        with picking_form.move_ids_without_package.new() as move:
+            move.product_id = self.productA
+            move.product_uom_qty = 5.0
+        picking = picking_form.save()
+        picking.action_confirm()
+
+        serial_wizard_dict = picking.move_ids_without_package.action_assign_serial()
+        serial_wizard = self.env[serial_wizard_dict['res_model']].with_context(serial_wizard_dict['context']).create({
+            'next_serial_number': '1',
+            'next_serial_count': 5,
+        })
+        serial_wizard.generate_serial_numbers()
```

Comment:
this is a clever way to auto-set the quantities, but you should still check that they are all set so that the test doesn't falsely pass due to the non-selected ml quantities being 0